code-challenges.test.js

```diff
@@ -14,39 +16,182 @@
 
-// b) Create the function that makes the test pass.
+// Declare new function letterFinder that takes in two arguments: array and string.
+const letterFinder = (array, string) => {
+  // Declare new variable letterArr to store the filtered array.
+  let letterArr = array.filter((value) => {
+    // Checks given array against any words that include the string character given.
+    return value.toLowerCase().includes(string.toLowerCase());
+  });
+  // Returns filtered array
+  return letterArr;
+};
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+
+// Test passed! I was having an issue where the return was an empty array. It was similar to the issue I was having with the first question, I wasn't telling the function to return the filtered result.
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
-
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+describe("fullHouse", () => {
+  it("takes in an array of 5 numbers and determines whether or not the array is a 'full house'.", () => {
+    const hand1 = [5, 5, 5, 3, 3];
+    // // Expected output: true
+    expect(fullHouse(hand1)).toEqual(true);
+
+    const hand2 = [5, 5, 3, 3, 4];
+    // // Expected output: false
+    expect(fullHouse(hand2)).toEqual(false);
+
+    const hand3 = [5, 5, 5, 5, 4];
+    // // Expected output: false
+    expect(fullHouse(hand3)).toEqual(false);
+
+    const hand4 = [7, 2, 7, 2, 7];
+    // Expected output: true
+    expect(fullHouse(hand4)).toEqual(true);
+  });
+});
+
+// ReferenceError: fullHouse is not defined
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 2 passed, 3 total
 
 // b) Create the function that makes the test pass.
+
+// I'm going to need to iterate over the array and run a conditional statement to check the values within the array. I'm not sure about the conditional logic though...
+
+// function fullHouse takes in an array and iterates over the array
+
+// I found this method online and wanted to give it a try but it doesn't really make sense to me honestly. I understand I need to iterate, but the logic of checking the values against each other without hardcoding the numbers is the hardest part for me right now.
```

Comment:
I appreciate that you are providing some of your research here.  The approach you looked at is pretty confusing at first.  To fully understand all of the steps on it, I would recommend console logging each line to see what each thing is doing if you want to understand this approach.  I feel like the approach you ended up with looks cleaner and is more understandable, so great work!